injectable/autowiring/autowired_decorator.py

```diff
@@ -69,10 +69,10 @@ def wrapper(*args, **kwargs):
             if parameter.name in bound_arguments:
                 continue
             dependency = parameter.annotation.inject()
-            if parameter.kind is parameter.KEYWORD_ONLY:
-                kwargs[parameter.name] = dependency
-            else:
+            if parameter.kind is parameter.POSITIONAL_ONLY:
                 args.append(dependency)
+            else:
+                kwargs[parameter.name] = dependency
```

Comment:
@craigminihan I've inverted this logic to always inject parameters as named args except for when the parameter is positional only. This will fix the issue and play nicely with Python 3.8's positional-only args too.

Reply:
@allrod5 nice work! I'll check out 3.4.1 shortly